internal/services/http.go

```diff
@@ -0,0 +1,66 @@
+package services
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// PrepareRequestForURL Prepares an HTTP GET request for a given URL.
+// It will apply additional HTTP headers if the host requires it.
+func PrepareRequestForURL(url string) (*http.Request, error) {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	const mozilla = "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0"
+
+	host := GetHost(url)
+	switch host {
+	case "aberlehome", "bettybossi", "chatelaine.com", "downshiftology.com", "marmiton", "natashaskitchen", "puurgezond", "reddit", "thekitchn", "thepalatablelife", "wellplated":
```

Comment:
To fix the 403 error for downshiftology.com, you can remove the `.com` from `downshiftology.com`. That is because GetHost returns `downshiftology`.

As for `chatelaine.com`, I receive a 200 with certain headers in Python using requests:
```python
import requests

headers = {
     'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0',
     'Cache-Control': 'max-age=0',
     'Pragma': 'no-cache',
     'DNT': '1',
}

requests.get('https://chatelaine.com/recipe/desserts/chocolate-buttercream-icing/', headers=headers)
<Response [200]>
```
However, I still get 403 when setting these same headers here. Further investigation will be required to determine what's preventing the request to succeed.